micro-ROS-Agent/micro-ros_entrypoint.sh

```diff
@@ -1,10 +1,19 @@
-. "/opt/ros/$ROS_DISTRO/setup.sh"
-. "/uros_ws/install/local_setup.sh"
+#!/bin/bash
 
 if [ "$ROS_LOCALHOST_ONLY" = "1" ] ; then
     export FASTRTPS_DEFAULT_PROFILES_FILE=/tmp/disable_fastdds_shm_localhost_only.xml
 else
     export FASTRTPS_DEFAULT_PROFILES_FILE=/tmp/disable_fastdds_shm.xml
 fi
 
-exec ros2 run micro_ros_agent micro_ros_agent "$@"
+if [ "$MICROROS_STATIC_SOURCE" = "1" ] ; then
+    export LD_LIBRARY_PATH=/uros_ws/install/micro_ros_agent/lib:/uros_ws/install/micro_ros_msgs/lib:/opt/ros/galactic/opt/yaml_cpp_vendor/lib:/opt/ros/galactic/lib/x86_64-linux-gnu:/opt/ros/galactic/lib:$LD_LIBRARY_PATH
```

Comment:
Although I see the point of the PR, this seems to be a change that will require quite a bit of maintenance. And can be an easy-to-forget mistake to forget to update the entry point.
Furthermore, wouldn't it be better to use the available $ROS_DISTRO variable rather than hardcoding the distro?

If this concept goes through, I'd suggest isolating the logic for static sourcing into a separate file, that could help a bit with the maintenance.

Reply:
+1

Reply:
Thank you for reviewing this patch.
I understand your point, and maybe it's not a widespread use case to justify this change. I can make this change only in my fork to avoid adding maintenances cost.